File: test.spritebuilder/Source/TDEngine/internalObject/MapStateInside.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package internalObject;

import Unit.Tower.Tower;
import Unit.Solder.Solder;
import Unit.Tower.Shell;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author vereshchagin
 */
public class MapStateInside extends MapStateInternal {
     
    
    public MapStateInside(List<Solder> unitsOnMap,List<Tower> towersOnMap,List<Shell> shells,boolean isEnd,boolean isWin,int money) {
      super();
        parseUnit(unitsOnMap);
        parseTower(towersOnMap);  
        parseShells(shells); 
        this.isEnd = isEnd;
        this.isWin = isWin;
        this.money = money;  
    }

    private void parseUnit(List<Solder> unitsOnMap) { 
      unit.clear();
        for(Solder x:unitsOnMap) 
            unit.add(x);   
    }

    private void parseTower(List<Tower> towersOnMap) { 
         towers.clear();
         for(Tower  x:towersOnMap) 
            towers.add(x);  
    }

    private void parseShells(List<Shell> shells) { 
         shells.clear();
        for(Shell   x:shells) 
          shells.add(x);     
    }

    
}

File: TowerDefence.spritebuilder/internalObject/MapStateInside.swift
//
//  MapStateInside.swift
//  TowerDefence
//
//  Created by koveres on 08.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation



public class MapStateInside : MapStateInternal {
    
    
    public init( isEnd1:Bool,  isWin1:Bool ) {
         super.init();
        
       
        var solders:[Solder] = EngAndFabric.inst_.soldeEng!.GetSoldersOnMap();
        var towers:[Tower] = EngAndFabric.inst_.towerEngine!.getTowers();
        var shells:[Shell] = EngAndFabric.inst_.towerEngine!.getShells();
        
        
        parseUnit(&(solders));
        parseTower(&(towers));
        parseShells(&(shells));
        isEnd = isEnd1;
        isWin = isWin1;
        money = EngAndFabric.inst_.money!.getMoney();
       
    }
    
    private  func  parseUnit( inout unitsOnMap:[Solder]) {
       
        for  x in unitsOnMap {
            unit.append(x);}
    }
    
    private  func  parseTower(inout towersOnMap:[Tower]) {
        
        for x in towersOnMap{
            towers.append(x);}
    }
    
    private  func  parseShells(inout shells:[Shell]) {
         for   x in shells{
            shells.append(x);}
    }
    
    
}
